crates/bevy_transform/src/hierarchy/hierarchy.rs

```diff
@@ -44,6 +44,18 @@ pub struct DespawnRecursive {
 }
 
 fn despawn_with_children_recursive(world: &mut World, entity: Entity) {
+    // first, make the entity's own parent forget about it
+    if let Ok(parent) = world.get::<Parent>(entity) {
+        if let Ok(mut children) = world.get_mut::<Children>(parent.0) {
+            children.0.retain(|c| *c != entity);
```

Comment:
```suggestion
            children.retain(|c| *c != entity);
```
<sub> _it's `DerefMut` all the way down_ </sub>

Reply:
Thanks!